dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/info/CallbackNetworkInfoProvider.kt

```diff
@@ -116,11 +116,29 @@ internal class CallbackNetworkInfoProvider(
 
     // region Internal
 
-    private fun getStrength(networkCapabilities: NetworkCapabilities): Int {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-            networkCapabilities.signalStrength
+    private fun resolveUpBandwidth(networkCapabilities: NetworkCapabilities): Long? {
+        return if (networkCapabilities.linkUpstreamBandwidthKbps > 0) {
+            networkCapabilities.linkUpstreamBandwidthKbps.toLong()
         } else {
-            Int.MIN_VALUE
+            null
+        }
+    }
+
+    private fun resolveDownBandwidth(networkCapabilities: NetworkCapabilities): Long? {
+        return if (networkCapabilities.linkDownstreamBandwidthKbps > 0) {
+            networkCapabilities.linkDownstreamBandwidthKbps.toLong()
+        } else {
+            null
+        }
+    }
+
+    private fun resolveStrength(networkCapabilities: NetworkCapabilities): Long? {
+        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q &&
+            networkCapabilities.signalStrength > NetworkCapabilities.SIGNAL_STRENGTH_UNSPECIFIED
```

Comment:
i'd use `!=` instead of `>` here, just because we don't know what the `SIGNAL_STRENGTH_UNSPECIFIED` value will be in the future.